server/src/main/java/org/elasticsearch/index/mapper/AbstractGeometryFieldMapper.java

```diff
@@ -197,15 +202,22 @@ public final void parse(DocumentParserContext context) throws IOException {
                 new IllegalArgumentException("Cannot index data directly into a field with a [script] parameter")
             );
         }
-        parser.parse(context.parser(), v -> index(context, v), e -> {
+
+        parser.parse(context.parser(), v -> index(context, v), (xContentParser, e) -> {
             if (ignoreMalformed()) {
                 context.addIgnoredField(fieldType().name());
+                onIgnoredMalformedValue(context, xContentParser);
             } else {
                 throw new MapperParsingException("failed to parse field [" + fieldType().name() + "] of type [" + contentType() + "]", e);
             }
-        });
+        }, ignoreMalformed() && context.isSyntheticSource());
```

Comment:
Rather than passing this boolean all the way down, can we not wrap the context parser here?  I think that would make this feel less fragile.

Reply:
The complexity here is because of arrays. For arrays, we will ignore only the ones that are malformed but keep the ones that are correctly parsed. So we need to generate one stored field value for each malformed element of the array.

Unfortunately array parsing is tricky because we accept values like `[0,0]` so array parsing is done on the point parser. 

That means as well we cannot wrap the parser here.

Reply:
Yeah. I tried wrapping here. It doesn't work well for the reasons @iverase mentioned. Partly because of the tricky formats we support and partly because we want to do `ignore_malformed` for each individual point.

I suppose the thing that I like the least is this whole "wrap the parser to support objects and it's always an object" thing. It just feels super leaky and weird. OTOH, what we're doing is tricky so maybe it has to leak to some degree.

Reply:
The value of `saveSyntheticSourceForMalformed` is always going to be the same for a given mapper, right? So I think my worries would be assuaged a bit if we added it as a parameter to the `PointParser` constructor rather than passing the same boolean value through half-a-dozen intervening functions for each document.

`ignoreMalformed` on `PointParser` is a bit weird anyway.  The Cartesian implementation uses it to ignore validation, but the GeoPoint version uses it to attempt normalisation (but then throws an error anyway if the values can't be normalised).